Sources/GRPCReflectionService/Documentation.docc/ReflectionServiceTutorial.md

```diff
@@ -0,0 +1,235 @@
+  the services of the Server that we want to be discoverable through reflection
+  and the version of the reflection service.
+- ``GRPCReflectionService/ReflectionService/init(fileDescriptorProtos:version:)``:
+  Receives as parameters the file descriptor protos of the proto files describing
+  the services of the Server that we want to be discoverable 
+  through reflection and the version of the reflection service.
+
+We recommend using the first one as it is more convinient. We will use it in this example as well.
+Before initialising the ReflectionService provider we need to generate the
+files containing the base64 encoded serialized file descriptor protos.
+
+
+### Generating the serialized file descriptor protos for the Server
+ The Server from this example uses the `GreeterProvider`, `EchoProvider` and the 
+ `ReflectionService` (v1 version). The associated proto files are located at 
+ `Sources/Examples/HelloWorld/Model/helloworld.proto`, 
+ `Sources/Examples/Echo/Model/echo.proto`, and 
+ `Sources/GRPCReflectionService/v1/reflection-v1.proto` respectively.
+
+ In order to generate the serialized file descriptor proto for the
+ `helloworld.proto`, you can run the following command:
+
+```sh
+$ protoc Sources/Examples/HelloWorld/Model/helloworld.proto \
+    --proto_path=Sources/Examples/HelloWorld/Model \
+    --plugin=./.build/debug/protoc-gen-grpc-swift \
+    --grpc-swift_opt=Client=false,Server=false,ReflectionData=true \
+    --grpc-swift_out=Sources/Examples/ReflectionService/Generated
+```
+
+Let's break the command down:
+ - The first argument that we are passing to `protoc` is the path 
+to the proto file we want to generate the serialized file descriptor proto
+for: [Sources/Examples/HelloWorld/Model/helloworld.proto][helloworld-proto].
+- The `proto_path` flag is the path to search for imports: 
+Sources/Examples/HelloWorld/Model.
+- The plugin we are using is `./.build/debug/protoc-gen-grpc-swift`.
+- The options for the `grpc-swift_opt` flag that we should set are:
+`Client=false,Server=false`, because we don't want to generate code
+and `ReflectionData=true` to signal that we want to generate a
+file containing the serialized file descriptor proto.
+- The `grpc-swift_out` flag is used to set the path of the directory
+where the generated file will be located in: Sources/Examples/ReflectionService/Generated.
+
+The commands for generating the binary files representing the serialized file descriptor 
+protos for the `Echo` and `Reflection` services are similar. The path of the output
+directory, the plugin and the options are the same. The path to search for imports
+is the parent directory of the proto file.
+
+The paths of the proto files are: 
+- [Sources/Examples/Echo/Model/echo.proto][echo-proto]
+- [Sources/GRPCReflectionService/v1/reflection-v1.proto][v1]
+
+Depending on the version of gRPCurl you are using you might need to use the v1alpha proto instead.
```

Comment:
We should link to gRPCurl here (https://github.com/fullstorydev/grpcurl) and we can be specific about the versions of gRPCurl which support v1 vs v1alpha (look at the release notes: https://github.com/fullstorydev/grpcurl/releases)